### DISCUSSION TITLE
[Unstable] Not all events updated with new CityModuleTags

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14124] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Since cities (and abandoned outposts) now use new module tags, not all events have been updated, causing some events to break in cities, like NPCs to spawn in the wrong places.

This affects both

`targetmoduletags`
`targetmoduletype`

Example: 
explosivemishap NPCs spawning in docking module because they're missing CityResearchModule

      <SpawnAction itemidentifier="ancientweapon" targettag="mishapweapon" spawnlocation="Outpost" targetmoduletags="researchmodule" SpawnPointTag="research" ignorebyai="true" />

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/e264d454-4bd7-44af-aef8-828d75f05fa6)

**The easier solution would be to save the city modules with the "normal" flags and keep the City location restriction**

This should solve the problem without hurting modding compatibility (including modded events), since they're only supposed to spawn in city to begin with
Meanwhile modders can feel free to use "Any" 

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/85595678-9702-4822-8f5d-bd2493a63626)

------------

_Honestly Dynamic Europa is almost the only mod leaning on custom locations and we are using custom flags and modules all the way. This is a bit of a headache in hindsight_

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

**Comment by Regalis11** [comment]
> This is a bit of a headache in hindsight

Be careful what you wish for. 😅

**Reply by ubertpendragon** [reply]
Yes I guess there arent enough people doing module/location mods for this to be relevant. 

Adding the old module flag alongside the new one + restricting to city would've worked without touching events themselves

**Comment by Regalis11** [comment]
Addressed this now by reverting the changes to the colony modules altogether. https://github.com/FakeFishGames/Barotrauma-development/commit/8dd240d94eadb1361c1bdb5f116ae0e21375f77d (commit in our private repo)
